java/adapter/jdbc/src/main/java/org/apache/arrow/adapter/jdbc/JdbcToArrow.java

```diff
@@ -215,4 +215,165 @@ public static VectorSchemaRoot sqlToArrow(ResultSet resultSet, JdbcToArrowConfig
     JdbcToArrowUtils.jdbcToArrowVectors(resultSet, root, config);
     return root;
   }
+
+
+  /*----------------------------------------------------------------*
+   |                                                                |
+   |          partial convert API                        |
+   |                                                                |
+   *----------------------------------------------------------------*/
+
+  /**
+   * For the given SQL query, execute and fetch the data from Relational DB and
+   * convert it to {@link ArrowVectorIterator}.
+   *
+   * @param connection Database connection to be used.
+   * @param query      The DB Query to fetch the data.
+   * @param allocator  Memory allocator
+   * @return Arrow Data Objects {@link ArrowVectorIterator}
+   * @throws SQLException Propagate any SQL Exceptions to the caller after closing any resources opened such as
+   *                      ResultSet and Statement objects.
+   */
+  public static ArrowVectorIterator sqlToArrowVectorIterator(
```

Comment:
I know you are are trying to keep consistency with the existing API but I think all these overrides are overkill.  I would prefer to just keep the methods that take a ResultSet, and probably just those with signatures:
(ResultSet, BaseAllocator)
(ResultSet, Config)
we should enforce that config requires an allocator.  Any other combinations can be used done with the Builder or added as convinience factor methods on Config.

Reply:
We should also probably mark all the methods above as deprecated since they encourage bad practice.